File: src/main/java/com/dafttech/terra/engine/gui/anchors/AnchorBottom.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

class AnchorBottom(val position: Float) extends GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer) {
    gObj.position.y = container.size.y * (1f - position) - gObj.size.y
  }
}

File: src/main/java/com/dafttech/terra/engine/gui/anchors/AnchorBottom.java
package com.dafttech.terra.engine.gui.anchors;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public class AnchorBottom extends GUIAnchor {
    public float position = 0;

    public AnchorBottom(float p) {
        position = p;
    }

    @Override
    public void applyAnchor(GUIObject object, GUIContainer container) {
        object.position.y = container.size.y * (1f - position) - object.size.y;
    }

}
